generate-default-operators/src/GenerateDefaultOperatorsModule.purs

```diff
@@ -0,0 +1,149 @@
+module GenerateDefaultOperatorsModule where
+
+import Prelude
+
+import Data.Argonaut.Core (Json)
+import Data.Argonaut.Decode as Data.Argonaut.Decode
+import Data.Array (foldMap, mapWithIndex)
+import Data.Array as Array
+import Data.Either (Either(..), either)
+import Data.Maybe (Maybe(..))
+import Data.Number (infinity)
+import Data.String (Pattern(..))
+import Data.String as Str
+import Data.String as String
+import Data.String.Regex as Regex
+import Data.String.Regex.Flags (noFlags)
+import Data.String.Regex.Unsafe (unsafeRegex)
+import Effect (Effect)
+import Effect.Aff (Aff, error, runAff_, throwError)
+import Effect.Class (liftEffect)
+import Effect.Class.Console as Console
+import Effect.Exception (throwException)
+import Foreign.Object (Object)
+import Foreign.Object as Foreign.Object
+import Node.Buffer as Buffer
+import Node.ChildProcess as ChildProcess
+import Node.ChildProcess as Exec
+import Node.ChildProcess.Types as Node.ChildProcess.Types
+import Node.Encoding (Encoding(..))
+import Node.FS.Aff (readdir, stat, writeTextFile)
+import Node.FS.Stats as FS
+import Node.Library.Execa as Node.Library.Execa
+import Node.Path (FilePath)
+import Node.Path as Path
+import Node.Process as Node.Process
+
+foreign import tmpdir :: String -> Effect String
+
+main :: Effect Unit
+main = runAff_ (either throwException mempty) do
+  cwdPath <- liftEffect Node.Process.cwd
+  tmpPath <- liftEffect $ tmpdir "purs-tidy-generate-default-operators-"
+  liftEffect $ Console.log $ "Working in " <> tmpPath
+
+  writeTextFile UTF8 (Path.concat [ tmpPath, "spago.yaml" ]) defaultSpagoYaml
+  writeTextFile UTF8 (Path.concat [ tmpPath, "package.json" ]) defaultPackageJson
+  liftEffect $ Console.log $ "running 'npm install'"
+  _ <- liftEffect $ ChildProcess.execSync' "npm install" (_ { cwd = Just tmpPath })
+
+  liftEffect $ Console.log $ "running 'spago ls packages --json'"
+  s <- liftEffect $ Buffer.toString UTF8 =<< Exec.execSync' "spago ls packages --json" (_ { cwd = Just tmpPath })
```

Comment:
The previous script had an explicit list of core/contrib largely due to these libraries changing very infrequently (and thus a conflict is rare), whereas a full package set is likely to change often. It's unlikely that a new version of tidy is going to be published with every operator change within the package set. I think would prefer not to commit to publishing operators for the full package set.

Reply:
can do `spago registry info PACKAGE --json` for each package

```
 /tmp/purs-tidy-generate-default-operators-kxYbzh  spago registry info z3 --json
{
  "location": {
    "githubOwner": "gbagan",
    "githubRepo": "purescript-z3"
  },
  "published": {
    "0.0.1": {
      "bytes": 6777,
      "hash": "sha256-9OoReWVBMDRquMGWR3yG98Xvz9PeMQLT7lFSAfFyH8U=",
      "publishedTime": "2023-05-05T16:45:43.000Z",
      "ref": "v0.0.1"
    },
    "0.0.2": {
      "bytes": 8734,
      "hash": "sha256-smw4c6eVA0AZOltlE1iVYDvD/KQxDjrz47UpZKwHg90=",
      "publishedTime": "2023-05-07T15:55:35.000Z",
      "ref": "v0.0.2"
    }
  },
  "unpublished": {}
}
```


filter only with `githubRepo` is `purescript-contrib` ....etc


-------------

```
#!/usr/bin/env node

const { execSync } = require('child_process');

// Get the package list from `spago ls packages --json`
const packagesJson = execSync('spago ls packages --json').toString();
const packages = JSON.parse(packagesJson);

// Array to store tuples of package names and their GitHub owners
const tuplePackagenameOwner = [];

// Loop through each package in the list
for (const packageName in packages) {
  try {
    // Get registry info for the current package
    const packageInfoJson = execSync(`spago registry info ${packageName} --json`).toString();
    const packageInfo = JSON.parse(packageInfoJson);

    // Extract the githubOwner if it exists
    const githubOwner = packageInfo?.location?.githubOwner || 'unknown';

    // Push tuple (packageName, githubOwner) into the array
    tuplePackagenameOwner.push([packageName, githubOwner]);
  } catch (error) {
    console.error(`Error fetching info for package: ${packageName}`, error.message);
  }
}

// Get unique GitHub owners
const uniqueGithubOwners = [...new Set(tuplePackagenameOwner.map(([, githubOwner]) => githubOwner))];

// Print the unique GitHub owners
console.log('Unique GitHub Owners:');
uniqueGithubOwners.sort().forEach(owner => console.log(owner));
```


```
AdaBeat
ConnorDillon
Ebmtranceboy
JordanMartinez
LiamGoodacre
MitchStevens
PureFunctor
Risto-Stevcev
Thimoteus
Unisay
acple
ad-si
ajnsit
albertprz
artemisSystem
athanclark
awakesecurity
bklaric
bodil
cakekindel
cdepillabout
citizennet
collegevine
coot
cprussin
crestonbunch
csicar
deadshot465
deemp
dewey92
easafe
ehrenmurdick
eikooc
ethul
f-f
f-o-a-m
fehrenbach
felixschl
flip111
flounders
garganscript
garyb
gbagan
grybiena
hdgarrood
i-am-the-slime
ilyakooo0
imsaravana369
jacereda
jamieyung
jeslie0
joellefkowitz
joneshf
jordanmartinez
jstoxrocky
justinwoo
kRITZCREEK
katsujukou
klntsky
krisajenkins
kritzcreek
lfarroco
lucasdicioccio
m-bock
m15a
maxdeviant
mcneissue
megamaddu
menelaos
mikesol
mjepronk
mlang
morganthomas
natefaubion
newlandsvalley
njaremko
nonbili
nsaunders
nwolverson
oreshinya
paf31
paluh
paulyoung
pete-murphy
prescientmoon
pujoheadsoft
purescript
purescript-concur
purescript-contrib
purescript-deprecated
purescript-grain
purescript-halogen
purescript-node
purescript-open-community
purescript-react
purescript-spec
purescript-web
qwbarch
rightfold
rnons
robertdp
rowtype-yoga
sharkdp
sigma-andex
sumew
tfc
the-dr-lazy
thomashoneyman
thought2
timdeputter
tonicebrian
truqu
tweag
ursi
userfriendanonymous
vibbits
wintechis
xc-jp
yukikurage
```


only

```
purescript-contrib
purescript-halogen
purescript-node
purescript-react
purescript-spec
purescript-web
```

?

Reply:
Previously it was only core and contrib. I don't think an upgrade to spago-next should re-evaluate that decision, so I would just go with that.